### PR TITLE
[cli] Add "image" optimization configuration support in `vercel.json`

### PR DESCRIPTION
**Pull request by theMosaad**
Continuation of https://github.com/vercel/vercel/pull/8217

### CODE REVIEW

**Comment by theMosaad** [comment]
The deployment creation API request is rejected with the new `images` property producing the below error:
```
> [debug] [2022-08-12T00:34:54.349Z] Error: Error: Schema verification failed
Error: Schema verification failed
    at createDeploy (/Users/mosaad/code/vercel/packages/cli/dist/index.js:237449:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at module.exports.__webpack_modules__.28640.exports.default (/Users/mosaad/code/vercel/packages/cli/dist/index.js:228735:22)
    at main (/Users/mosaad/code/vercel/packages/cli/dist/index.js:234836:20)
Error! Invalid vercel.json - should NOT have additional property `images`. Please remove it.
View Documentation: https://vercel.com/docs/configuration
```

**Comment by TooTallNate** [comment]
Thanks for the follow-up here. I think this logic should go directly into the `vercel build` command though, so that all projects will be able to utilize it.

File: packages/cli/src/commands/build.ts [hunk 1]
```diff
@@ -549,7 +549,153 @@ async function doBuild(
     builds: builderRoutes,
   });
 
-  const mergedImages = mergeImages(buildResults.values());
+  let images: BuildResultV2Typical['images'] = undefined;
+  if (vercelConfig?.images) {
+    images = vercelConfig?.images;
+  }
```
Comment (styfle) [inline packages/cli/src/commands/build.ts]:
```suggestion
  const images = vercelConfig?.images
```

**Review by styfle — COMMENTED** [review]
Looks great, just needs a test 👍 

You can add one here:

https://github.com/vercel/vercel/blob/7523e39f18f6c2994aa8a46e243364f6f010dabd/packages/cli/test/unit/commands/build.test.ts

File: packages/cli/test/unit/commands/build.test.ts [hunk 2]
```diff
@@ -894,6 +894,31 @@ describe('build', () => {
...
+    const cwd = fixture('images');
+    const output = join(cwd, '.vercel/output');
+    try {
+      process.chdir(cwd);
+      const exitCode = await build(client);
+      expect(exitCode).toEqual(0);
+
+      // `config.json` includes "images" from `vercel.json`
+      const configJson = await fs.readJSON(join(output, 'config.json'));
+      console.log({ configJson });
```
Comment (styfle) [inline packages/cli/test/unit/commands/build.test.ts]: Lets' remove the console.log

```suggestion
```

**Review by styfle — APPROVED** [review]
Perfect, thanks!

**Comment by TooTallNate** [comment]
This was merged in https://github.com/vercel/vercel/pull/8566. Thank you @theMosaad!

**Pull request closed without merging**